### PR TITLE
Fix public flag when creating intersection error type

### LINKED ISSUE #36203: Non-public types are accessible outside the module
**Description:**
> $Subject

**Steps to reproduce:**

1. Create one module : `module1` (`bal new module1`) and add the following to the `main.bal` : 
```ballerina
import ballerina/io;

// Level 1
public type Error distinct error;

// Level 2
public type ClientError distinct Error;
public type ListenerError distinct Error;
public type PayloadBindingError distinct Error;

// Level 3 - Internal Error Types
type PayloadBindingClientError ClientError & PayloadBindingError;
type PayloadBindingListenerError ListenerError & PayloadBindingError;

public function main() {
    io:println("Hello, World!");
}
```

2. Pack the module (`bal pack`) and push it to the local central (`bal push --repository=local`)

3. Create another module : `module2` (`bal new module2`)

4. Add the `module1` dependency to `module2/Ballerina.toml`:
```toml
[[dependency]]
org="user"
name="module1"
repository="local"
version="0.1.0"
```

5.  Add the following to the `main.bal` : 
```ballerina
import ballerina/io;
import user/module1;

public function main() {
    error err = error module1:PayloadBindingClientError("Hello, World!");
    io:println(err.message());
}
```

6. Run `module2` (`bal run module2`). There will be no compiler errors saying that `PayloadBindingClientError` is not accessible.

**Affected Versions:**
Ballerina SwanLake 2201.1.0 - RC1

> Note : 
There is no compiler errors even when the intersection types are non-public. But when there is no intersection, it is giving an error

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #36203 

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 1]
```diff
@@ -2421,8 +2421,8 @@ private BIntersectionType createIntersectionErrorType(IntersectableReferenceType
                                                           BSymbol owner, Location pos) {
 
         BTypeSymbol intersectionTypeSymbol =
-                Symbols.createTypeSymbol(SymTag.INTERSECTION_TYPE, Flags.asMask(EnumSet.of(Flag.PUBLIC)), Names.EMPTY,
-                        pkgId, null, owner, pos, VIRTUAL);
+                Symbols.createTypeSymbol(SymTag.INTERSECTION_TYPE, 0, Names.EMPTY, pkgId, null, owner, pos,
+                        VIRTUAL);
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:2425]: Can we make this a single line, not sure if it reaches the 120 charactor limit.

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:2425]: It does reach 120.

**Review by rdhananjaya — APPROVED** [review]

File: tests/jballerina-unit-test/src/test/resources/test-src/error/error-negative-project/modules/module1/module1.bal [hunk 2]
```diff
@@ -0,0 +1,31 @@
+public type E1 distinct error;
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/error/error-negative-project/modules/module1/module1.bal:1]: Shall we add license header here?

**Review by pcnfernando — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java [hunk 3]
```diff
@@ -2421,8 +2421,7 @@ private BIntersectionType createIntersectionErrorType(IntersectableReferenceType
                                                           BSymbol owner, Location pos) {
 
         BTypeSymbol intersectionTypeSymbol =
-                Symbols.createTypeSymbol(SymTag.INTERSECTION_TYPE, Flags.asMask(EnumSet.of(Flag.PUBLIC)), Names.EMPTY,
-                        pkgId, null, owner, pos, VIRTUAL);
+                Symbols.createTypeSymbol(SymTag.INTERSECTION_TYPE, 0, Names.EMPTY, pkgId, null, owner, pos, VIRTUAL);
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java:2424]: False positive codecov comment

**Pull request merged**